device_doctor/bin/main.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+
+import 'package:device_doctor/src/device.dart';
+import 'package:device_doctor/src/health.dart';
+import 'package:device_doctor/src/config.dart';
+
+const String actionFlag = 'action';
+const String deviceOSFlag = 'deviceOS';
+const String helpFlag = 'help';
+
+String _action;
+String _deviceOS;
+
+/// Manage healthcheck, cleanup and restart for devices configured in
+/// config.yaml.
+///
+/// Usage:
+/// dart main.dart --action <healthcheck|cleanup|restart>
+Future<void> main(List<String> args) async {
+  final ArgParser parser = ArgParser();
+  parser
+    ..addOption('$actionFlag')
+    ..addOption('$deviceOSFlag')
+    ..addFlag('$helpFlag');
+
+  if (!_checkArgs(parser, args)) {
+    stdout.write('\nRequired flags:\n'
+        '--$actionFlag Supported actions are healthcheck, recover, and cleanup.\n'
+        '--$deviceOSFlag Supported device OS: android or ios.\n');
+    exit(1);
+  }
+
+  Config.initialize(_deviceOS);
+
+  switch (_action) {
+    case 'healthcheck':
+      final Map<String, HealthCheckResult> deviceChecks = await devices.checkDevices();
+      final bool hasHealthyDevices = deviceChecks.values.where((HealthCheckResult r) => r.succeeded).isNotEmpty;
+      if (!hasHealthyDevices) {
+        throw StateError('No healthy $_deviceOS device is available');
+      }
+      if (_deviceOS == 'ios') {
```

Comment:
This is a recovery step rather than health check. The health check will only include the validation of whether or not a device is showing a system dialog.